### ISSUE TITLE
Unparsable control characters in graph.json breaks cached-resolution feature

### ISSUE THREAD

**Issue by littlenag** [sbt/sbt#4117]
(See the guidelines for contributing, linked above)

## steps

Ensure you have enabled cached-resolution for your build: 

```
updateOptions := updateOptions.value.withCachedResolution(true)
```

As followed from: https://www.scala-sbt.org/1.x/docs/Cached-Resolution.html

We then run sbt.

## problem

For some of our projects the generated graph.json file will contain content that looks like the following:

```
"licenses":[["Copyright © 2016 Vast.com","http:\/\/localhost"]]
```

The relevant part is the "©" character. When parsing the graph.json file the Jawn parser will throw an error similar to:

```
[error] Unable to parse mini graph: /Users/luka/.sbt/0.13/dependency/module/static/com.vast.common/common-rest-client_2.11/7.0.0_temp-resolve-8e9431e905f008b91f08c051d2774e80298f9769/graphs/graph.json scala.pickling.PicklingException: failed to parse "/Users/luka/.sbt/0.13/dependency/module/static/com.vast.common/common-rest-client_2.11/7.0.0_temp-resolve-8e9431e905f008b91f08c051d2774e80298f9769/graphs/graph.json" as JSON: control char (-62) in string got ￂ (line 1, column 8416) at 
sbt.serialization.json.JSONPickle$.fromFile(JSONPickleFormat.scala:75) at 
sbt.serialization.SerializedValue$.fromJsonFile(SerializedValue.scala:70) at 
sbt.serialization.SerializationFunctions$class.fromJsonFile(SerializationFunctions.scala:18) at 
sbt.serialization.package$.fromJsonFile(package.scala:9)
```

Which matches the behavior we see in code here https://github.com/sbt/serialization/blob/v0.1.1/serialization/src/main/scala/sbt/serialization/json/JSONPickleFormat.scala

## expectation

I would expect build not to fail in the presence of control characters. Presumably the character needs to be properly escaped, or the parser updated to handle control characters. If neither of those are possible, then the relevant code should refuse to handle un-marshallable text and fail early enough so that we can correct the issue.

## notes

sbt version: 0.13.16 (but issue has appeared in nearly every 0.13.x version)

**Comment by littlenag** [comment]
For now our work-around is to disable cached resolution.

**Comment by dukejansen** [comment]
This continues to be an ongoing frustration.

**Comment by eed3si9n** [comment]
Do you set JAVA_OPTS environment variable? If not would

```sh
$ export JAVA_OPTS=-Dfile.encoding=UTF-8
```

help?

**Comment by DanilSuits** [comment]
No, exporting those JAVA_OPTS does not improve the observed behavior.

**Comment by eed3si9n** [comment]
Does this happen on sbt 1.x? `sbt.serialization` was using Scala Pickling, but we no longer use that stuff.
